### ISSUE TITLE
2.0-M1-groovy-2.5 Problem with TemporaryFolder

### ISSUE THREAD

**Issue by SteveHigham** [spockframework/spock#1066]
I have a number of tests which use statements TemporaryFolders as follows:
@Rule final TemporaryFolder testFolder = new TemporaryFolder ()

These work as expected under Spock 1.3-groovy-2.5 / JDK 1.8 but fail with an IllegalStateException under Spock 2.0-M1-groovy-2.5 / JDK 11. I have also tried:
@Rule TemporaryFolder testFolder

This fails with a NullPointerException. If I initialise it in a setup () method it fails with an IllegalStateException in the test saying that the folder has not yet been created.

The tests are being run under Gradle 6.0.1 on a Mac.

**Comment by szpak** [comment]
Please consider the ["Breaking changes"](Breaking Changes) part in the release notes. Spock 2.0-M1 uses JUnit Platform which doesn't support [*] JUnit 4 anymore. You may want to consider the spock-junit4 module.

[*] - there is a [limited experimental support](https://junit.org/junit5/docs/current/user-guide/#migrating-from-junit4-rule-support) for (some of) JUnit 4 rules in JUnit Platform itself (`@EnableRuleMigrationSupport`), but it shouldn't be needed in Spock.

**Comment by SteveHigham** [comment]
Thanks @szpak - bit of brain fade - should of checked the notes first. Your comments helped me resolve my issue so I'm closing this issue.

It looks like the JUnitPlatform solution will be to use @TempDir but this is still an experimental feature. Therefore I have regressed to Spock 1.3-groovy-2.5. I've discovered that getting my tests to run under JDK 11 was as simple as removing the final modifier so I now have:
@rule /*final*/ TemporaryFolder testFolder = new TemporaryFolder ()

I will move back to 2.0 when it is released and the JUnit 5 @TempDir feature is released.

**Comment by szpak** [comment]
Btw, in my projects adding the `spock-junit4` module - which under the hood wraps JUnit 4's rules into Spock extensions - worked out of the box with `@TemporaryFolder` (still being executed as JUnit 5 tests). Have you encounter any problem with that approach (waiting for `@TestDir` to be a part of the stable API)?

**Comment by SteveHigham** [comment]
I think I'm leaving things as they are for now as it's all working and I don't want to upgrade to Spock 2.0 until it's stable. I note that the spock-junit4 module is only available for Spock 2.0.
I will pick this up if Spock 2.0 is released before @tempdir is stable.
Thanks again for the advice :-)

**Comment by szpak** [comment]
Sure, I don't state that Spock 2.0-M1 is production ready, but it's good to make a try to detect potential errors and report them (as you did!). It's great to hear (as I understood you correctly) that you didn't have any other problems with your tests and Spock 2.0-M1.

**Comment by SteveHigham** [comment]
Hi @szpak,

Now I have my JDK 11 upgrade working and stable, I have come back to your suggestion to try testing against Spock 2. My tests have all worked without issue by using the spock-junit4 module as you suggested. The numbers are as follows.

All my tests are run from gradle. I have 316 unit tests and 57 integration tests. The unit tests are all within buildSrc. The integration tests all test one of my custom gradle tasks by using the gradle runner. They are spread across various modules and executed in parallel.

The tests were run on my iMac (Catalina 10.15.2) and my laptop (Ubuntu 18.04.3 LTS). On the iMac I tested against Oracle Java 1.8.0.211 and  Oracle Java 11.0.5. On the laptop I tested against Oracle Java 1.8.0_181 and OpenJDK 11.0.5.

Cheers, Steve
